statesync/reactor.go

```diff
@@ -350,6 +349,7 @@ func (r *Reactor) processSnapshotCh() {
 		select {
 		case envelope := <-r.snapshotCh.In():
 			if err := r.handleMessage(r.snapshotCh.ID(), envelope); err != nil {
+				r.Logger.Error("failed to process envelope", "ch_id", r.snapshotCh.ID(), "envelope", envelope, "err", err)
```

Comment:
Shouldn't this be logged by the shim? We can just wrap the error with additional context.

Reply:
Well I'd prefer to have it here for two reasons:

1. The shim is being removed, although the router will replace it, I don't know what it'll look like.
2. The shim isn't used in reactor tests and so when testing with verbosity enabled, these logs are very helpful (they helped me a lot when debugging). If you move them up to the shim, you lose them completely.

I suppose we could have more context-dependent logs in `handleMessage`. Would that work for you?

Reply:
Good points. Yeah, I'm not really sure how we'll be doing this in the router, and whether it makes sense to log errors there (maybe only if they end up disconnecting the peer or something). Let's defer it for now.